### PR TITLE
openjdk8: mips* support

### PR DESCRIPTION
**Pull request by fosslinux**
Builds for all mips* platforms, using the zero backend. A mips port
properly is in process but has been "getting close to ready" for years
now, so we are using the much more tested zero backend.

mipshf had a branch jump error (mips hates super large branch jumps).
This is a known bug that is not going to be fixed very easily. To
workaround this, I reduced the jump by ~400 lines of assembly by
compiling with -Os, enough to make the jump close enough.

The mips support patch was taken from debian, I left mips64 in there because it might be useful in the future, we don't lose anything by having it there. (I debated putting the whole patch with all the extra archs in there, but it looked a bit unwieldy then)

cc @pullmoll recent committer and mips expert ;)
cc @Vaelatern maintainer

openjdk11 coming up

### CODE REVIEW

**Comment by Vaelatern** [comment]
Shouldn't be able to break other platforms, so I'm OK with this.

File: srcpkgs/openjdk8/template [hunk 1]
```diff
@@ -186,11 +186,17 @@ do_configure() {
 	CXXFLAGS=${CXXFLAGS/-D_FORTIFY_SOURCE=2/}
 
 	# force ELFv2 for ppc64 just in case
+	# with normal -O2 on mipshf there is a too-large jump which causes the
+	# assembler to fail. optomising for size means the jump is made smaller
```
Comment (ericonr) [inline srcpkgs/openjdk8/template:190]: "optimising"

And I think it would be best if the comments were right above their respective archs, since there are different special cases here.

**Comment by fosslinux** [comment]
@Vaelatern Is it OK just to have a link to the debian URL in the header of the patch?

**Comment by fosslinux** [comment]
Addressed.

**Comment by fosslinux** [comment]
Ping
